Kernel/FileSystem/Ext2FileSystem.cpp

```diff
@@ -1209,6 +1209,7 @@ KResult Ext2FSInode::add_child(Inode& child, const StringView& name, mode_t mode
     if (result.is_error())
         return result;
 
+    VERIFY(populate_lookup_cache());
```

Comment:
Please don't do effectful things inside `VERIFY` as it makes it much harder to see what's happening.

Also, if populating the lookup cache causes an error, we should propagate that error to the caller instead of panicking the kernel :)

Reply:
Thanks!

Should I modify `populate_lookup_cache()` to return the error? Right now it just returns a bool.

Also there's one place where the result of `populate_lookup_cache()` is ignored: `Ext2FSInode::directory_entry_count()`, so probably that should propagate it too? It's also called in `Ext2FSInode::lookup()`, but that doesn't have a way of returning errors except an empty pointer.

Reply:
Thanks! I've modified populate_lookup_cache() to return a KResult, and the error is propagated out everywhere it's called.